### DISCUSSION TITLE
Tuya unknown MCU, no TuyaReceived

### DISCUSSION THREAD

**Discussion by sander1988** [arendst/Tasmota#15006] [Q&A]
### PROBLEM DESCRIPTION
I have flashed a device (Wakeup light) with Tasmota. The main chip running on a TYWE3S, this part works.

The MCU is a **CA51F253L3**, I can't find much about it. But I assume it should be compatible with the TuyaMCU protocol (maybe I'm wrong). Unfortunately  the MCU isn't detect (no "TuyaReceived"). I think I have tried everything over the last few days, but can't figure it out.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): TYWE3S
- [x] Tasmota binary firmware version number used: 9.3.0
  - [ ] Pre-compiled
  - [x] Self-compiled
- [x] Flashing tools used: esptool and web.
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
18:30:09.763 RSL: RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"2272":"Tuya Tx"},"GPIO2":{"0":"None"},"GPIO3":{"2304":"Tuya Rx"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}
```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
No rules
```
- [x] Provide the output of this command: `Status 0`:
```lua
18:31:01.361 RSL: STATUS = {"Status":{"Module":54,"DeviceName":"Wakeup light","FriendlyName":["Wakeup light"],"Topic":"A052C1","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0}}
18:31:01.367 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"hcs","OtaUrl":"","RestartReason":"Software/System restart","Uptime":"0T00:01:13","StartupUTC":"2022-02-28T17:29:48","Sleep":50,"CfgHolder":4617,"BootCount":18,"BCResetTime":"2022-02-27T13:41:51","SaveCount":38,"SaveAddress":"F5000"}}
18:31:01.372 RSL: STATUS2 = {"StatusFWR":{"Version":"9.3.0(tasmota)","BuildDateTime":"2022-02-27T14:20:35","Boot":7,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"350/699"}}
18:31:01.377 RSL: STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":4,"MqttLog":0,"SysLog":2,"LogHost":"x.x.x.100","LogPort":10514,"SSId":["xxxxxx",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008001","2805C8000100060000005A00000000000000","00000080","0020E000","00000000"]}}
18:31:01.386 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":540,"Free":460,"Heap":28,"ProgramFlashSize":1024,"FlashSize":2048,"FlashChipId":"1540C8","FlashFrequency":40,"FlashMode":3,"Features":["00000809","088004C6","04008001","00000047","010001C0","40000900","00000000","00001000","00000000"],"Drivers":"1,2,3,4,8,10,16,27,29,45","Sensors":"1,2,3,5"}}
18:31:01.391 RSL: STATUS5 = {"StatusNET":{"Hostname":"A052C1-4801","IPAddress":"x.x.x.140","Gateway":"x.x.x.1","Subnetmask":"255.255.255.0","DNSServer":"x.x.x.1","Mac":"xxxxxxx","Webserver":2,"WifiConfig":4,"WifiPower":17.0}}
18:31:01.397 RSL: STATUS7 = {"StatusTIM":{"UTC":"2022-02-28T17:31:01","Local":"2022-02-28T18:31:01","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":99}}
18:31:01.401 RSL: STATUS10 = {"StatusSNS":{"Time":"2022-02-28T18:31:01"}}
18:31:01.407 RSL: STATUS11 = {"StatusSTS":{"Time":"2022-02-28T18:31:01","Uptime":"0T00:01:13","UptimeSec":73,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"xxxxxx","BSSId":"xxxxxx","Channel":11,"RSSI":80,"Signal":-60,"LinkCount":1,"Downtime":"0T00:00:03"}}}
```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
...
18:31:31.761 WIF: Checking connection...
18:31:38.419 TYA: Send "55aa00000000ff"
18:31:48.441 TYA: Send "55aa001c00080116021c121f3001ba"
18:31:49.441 TYA: Send "55aa00000000ff"
18:31:51.763 WIF: Checking connection...
18:32:00.430 TYA: Send "55aa00000000ff"
18:32:11.422 TYA: Send "55aa00000000ff"
18:32:11.772 WIF: Checking connection...
18:32:22.407 TYA: Send "55aa00000000ff"
18:32:31.798 WIF: Checking connection...
18:32:33.418 TYA: Send "55aa00000000ff"
18:32:44.450 TYA: Send "55aa00000000ff"
18:32:48.411 TYA: Send "55aa001c00080116021c12203001bb"
18:32:51.778 WIF: Checking connection...
18:32:55.439 TYA: Send "55aa00000000ff"
18:33:06.412 TYA: Send "55aa00000000ff"
...
```

### TO REPRODUCE
* Used the docs: https://tasmota.github.io/docs/TuyaMCU/ .
  * WebLog 4
  * Module set to 54
* I tried every known pin config (none work):
  * GPIO1 = Tuya Tx, GPIO3 = Tuya Rx
  * GPIO1 = Tuya Rx, GPIO3 = Tuya Tx
  * GPIO13 = Tuya Tx, GPIO15 = Tuya Rx
  * GPIO13 = Tuya Rx, GPIO15 = Tuya Tx
* I tested with both baud rates (SetOption97).
* I follow the PCB traces (brown and white wires) and I think it's using the common pins (GPIO1 = Tuya Tx, GPIO3 = Tuya Rx). These pins are connect with the MCU (with a 1K resistor for both traces).

Any suggestions in things to test? Did I miss something? I tried everything the last couple of days. Any suggestion or help is appreciated.

### EXPECTED BEHAVIOUR
No 'TuyaReceived', the message/response I would expect here. Only 'TYA: Send' in the weblog.

### SCREENSHOTS
![IMG_0467_2](https://user-images.githubusercontent.com/6334319/156032472-affa9a3c-4bab-458d-bc6e-9920481f8b4e.jpg)

### ADDITIONAL CONTEXT
* The MCU has seperate pads on the PCB marked TX and RX. I did run a tester (script which test all known baud rates to search for console). But the MCU doesn't show any output.
* I found some info about a previous revision of the device. But the PCB is different in this revision: https://www.reddit.com/r/tasmota/comments/kkloqc/tasmota_92_wake_up_light_advice_needed/
* GPIO1 = Tuya Tx is marked red in the image.
* GPIO3 = Tuya Rx is marked green in the image.
* The MCU itself is working/booted. LEDs to change when I click on the buttons of the devices.

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by CraigHutchinson** [comment]
Bit late, maybe you solved this but looks like Baud rate is set at 115200, most Tuya MCUs use baud rate to 9600 bps

**Comment by MrdUkk** [comment]
the MCU is very very known. at least for me :) i have flasher for it (look at my repos).  the internet have its SDK, compiler and anything to develop :) the chip is very very cool
